api.ts

```diff
@@ -1404,10 +1414,16 @@ export function makeSyncTable<
 }: SyncTableOptions<K, L, ParamDefsT, SchemaDefT>): SyncTableDef<K, L, ParamDefsT, SchemaT> {
   const {getSchema: getSchemaDef, entityName} = dynamicOptions;
   const {execute: wrappedExecute, ...definition} = maybeRewriteConnectionForFormula(formula, connectionRequirement);
-  if (schemaDef.identity) {
-    schemaDef.identity = {...schemaDef.identity, name: identityName || schemaDef.identity.name};
-  } else if (identityName) {
-    schemaDef.identity = {name: identityName};
+  // We don't fail on a missing identityName because the legacy functions don't set it.
+  if (identityName) {
```

Comment:
I'm not sure why I get why all the below logic is newly conditional on this? Naively I would think that we just want to add the validation about the names matching.

Reply:
The old code was already a no-op if identityName was falsy, I just thought this would be easier to read.

Reply:
Ah, that's subtle, I see it now, yeah you're right.